cpp/src/svm/svr_impl.cuh

```diff
@@ -40,16 +40,17 @@ namespace ML {
 namespace SVM {
 
 template <typename math_t>
-void svrFit(const raft::handle_t& handle,
-            math_t* X,
-            int n_rows,
-            int n_cols,
-            math_t* y,
-            const SvmParameter& param,
-            raft::distance::kernels::KernelParams& kernel_params,
-            SvmModel<math_t>& model,
-            const math_t* sample_weight)
+void svrFitX(const raft::handle_t& handle,
+             const MLCommon::Matrix::Matrix<math_t>& matrix,
```

Comment:
Please use `svrFitX` and `svrFitXSparse` instead of exposing additional matrix vocabulary types. Raw pointers should be fine since this is primarily being invoked by the cython/python layer.